### ISSUE TITLE
False-positive for D208 "Docstring is over-indented" for code sample at the end of module docstring

### ISSUE THREAD

**Issue by thomkeh** [charliermarsh/ruff#986]
I don't know if this appears anywhere else, but I observed it in a module docstring.

This:
```python
"""Module docstring.

.. code:: python
   a = f()
   assert a
"""
```
gives the error:
```
Found 1 error(s).
main.py:5:1: D208 Docstring is over-indented
```
whereas this:
```python
"""Module docstring.

.. code:: python
   a = f()
   assert a

More documentation.
"""
```
gives no error.

I can understand though if this is a wont-fix because I can just turn off the `D208` error code.

**Comment by charliermarsh** [comment]
I think this can comfortably be considered a bug because it deviates from the `pydocstyle` behavior.
